lib/elements/reinforcement/straight_longitudinal.rb

```diff
@@ -0,0 +1,80 @@
+require 'elements/reinforcement/base'
+require 'elements/reinforcement/straight_longitudinal_layer'
+require 'errors/reinforcement/empty_layers'
+
+module Elements
+  module Reinforcement
+    class StraightLongitudinal < Base
+      attr_accessor :layers
+
+      def initialize(z_base:, distribution_direction:, direction: 1)
+        @z_base = z_base
+        @direction = direction
+        @layers = []
+        @distribution_direction = distribution_direction
+      end
+
+      def modify_z_base(z_base:)
+        @z_base = z_base.to_f
+        update_z_base_of_layers unless @layers.empty?
+      end
+
+      def add_layer(start_location:, end_location:, number_of_rebars:, rebar:)
+        start_location.value_3 = @z_base
+        end_location.value_3 = @z_base
+        id = @layers.empty? ? 1 : @layers.last.id
+        @layers << Elements::Reinforcement::StraightLongitudinalLayer.new(
+          start_location:,
+          end_location:,
+          number_of_rebars:,
+          rebar:,
+          id:,
+          distribution_direction: @distribution_direction
+        )
+
+        @layers.last
+      end
```

Comment:
can you take a look to [--> our wiki <-- Elements::Reinforcement::StraightLongitudinal #add_layer section](https://www.notion.so/pradaing/Reinforcement-ff39ec7738fe4692a807ec9eaa1063ab#a3e3a967c71046f499fd25792e603d75)